includes/db/Database.php

```diff
@@ -3521,23 +3521,28 @@ protected function executeAndProfileQuery( $sql, $fname, $isMaster ) {
 	 * at the rate defined in self::QUERY_SAMPLE_RATE.
 	 *
 	 * @param string $sql the query
-	 * @param ResultWrapper|resource $ret database results
-	 * @param string $fname the function name
+	 * @param ResultWrapper|resource|bool $ret database results
+	 * @param string $fname the name of the function that made this query
 	 * @param bool $isMaster is this against the master
 	 * @return void
 	 */
 	protected function logSql( $sql, $ret, $fname, $elapsedTime, $isMaster ) {
 		global $wgDBcluster;
 
-		if ( $ret === false ) {
-			return;
+		if ( $ret instanceof ResultWrapper ) {
+			$num_rows = $ret->numRows();
+		} elseif ( is_resource( $ret ) ) {
+			$num_rows = mysql_num_rows( $ret );
+		} else {
+			// INSERT, UPDATE, DELETE, DROP queries do not report num_rows
```

Comment:
Although we can grab `mysql_affected_rows` from the connection resource, if available in this scope.